### ISSUE TITLE
How does sails default to serve assets/index.html?

### ISSUE THREAD

**Issue by jedd-ahyoung** [balderdashy/sails#2493]
I'm having an issue where I'd like to have a catchall route to point to my index.html in assets (an Angular SPA).

In config/routes.js, removing the default root route handler (`'/'`) will cause Sails to serve index.html from `assets/` by default. However, I can't find a good way to do this for a catchall route (as I have to find a way to manually serve from `.tmp/public`).

I've tried a few things but nothing's elegant, so I decided to consult the Sails core to see if I could find where the route resolves to index.html for the default route. I wasn't able to track it down, though. Can anybody shed some light into this?

My thinking is that if I can find where this happens, I can either create a route configuration key (like "asset" or something) to leverage this, or maybe add a hook somewhere to handle it. Something like follows:

    module.exports = {
		'/*': {
			asset: 'index.html',
			skipAssets: true
		},
	};

Thanks! Please let me know if this would be better posted on SO.

**Comment by tanertopal** [comment]
I think the easiest way would be to prefix your api with something like /api/something and then use the 404 to serve the index file for all 404 which do not go to /api/something

**Comment by sailsbot** [comment]
Thanks for posting, @jedd-ahyoung.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by edwardhotchkiss** [comment]
```javascript
'*': {
  target: '/',
  skipAssets: true
}
```
